From 901651a84f6a08daff00e3f2cef81d5f5ce51c34 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 18 Nov 2010 10:27:29 -0600
Subject: [PATCH] omap: beagle: set core voltage to 1.35 for xM

Based of a patch by: Max Galemin posted to the beagleboard mailing list

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/mach-omap2/board-omap3beagle.c |    8 ++++++++
 include/linux/i2c/twl.h                 |    2 ++
 2 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/board-omap3beagle.c b/arch/arm/mach-omap2/board-omap3beagle.c
index e981bb8..83a8f97 100644
--- a/arch/arm/mach-omap2/board-omap3beagle.c
+++ b/arch/arm/mach-omap2/board-omap3beagle.c
@@ -501,6 +501,14 @@ static int beagle_twl_gpio_setup(struct device *dev,
 		/* TWL4030_GPIO_MAX + 0 == ledA, EHCI nEN_USB_PWR (out, active low) */
 		gpio_request(gpio + TWL4030_GPIO_MAX, "nEN_USB_PWR");
 		gpio_direction_output(gpio + TWL4030_GPIO_MAX, 1);
+
+		#define SET_VDD1_VSEL_1_35_V    0x3C
+
+		/* See TPS65950 TRM (SWCU050F) "5.4.10.1.1 Direct Control" for details. */
+		twl_i2c_write_u8(TWL4030_MODULE_PM_RECEIVER,
+				SET_VDD1_VSEL_1_35_V, TWL4030_VDD1_VSEL);
+
+		printk(KERN_INFO "OMAP3 Beagle xM: Set VDD1 to 1.35");
 	}
 	else {
 		/* DVI reset GPIO is different between revisions */
diff --git a/include/linux/i2c/twl.h b/include/linux/i2c/twl.h
index 7a5e4ab..c5ef720 100644
--- a/include/linux/i2c/twl.h
+++ b/include/linux/i2c/twl.h
@@ -626,6 +626,8 @@ int twl4030_sih_setup(int module);
 #define TWL4030_VAUX2_DEDICATED		0x1E
 #define TWL4030_VAUX3_DEV_GRP		0x1F
 #define TWL4030_VAUX3_DEDICATED		0x22
+#define TWL4030_VDD1_VSEL		0x5E
+#define TWL4030_VDD2_VSEL		0x6C
 
 static inline int twl4030charger_usb_en(int enable) { return 0; }
 
-- 
1.7.1

